Fix for 64bit Py3.3 & 3.4 libpq build issues

jerickso opened the pull request:
Copy over 64bit rc* binary files to the 32bit location since for some reason when building libpq, the postgres build setup chooses to use the 32bit version location of rc.exe.  This causing the program to fault and never build libpq.

jerickso committed:
Fix 64bit Py3.3 & 3.4 libpq build issues

Copy over 64bit rc* binary files to the 32bit location since for some reason
when building libpq, the postgres build setup chooses to use the 32bit version
location of rc.exe.  This causing the program to fault and never build
libpq.

jerickso committed:
on_success to deploy built files if on psycopg2 account

To assisting debugging when not on the main psycopg2 account, add a check
for the REMOTE_KEY environment variable, creating a dummy value for the
key if it doesn't exist.  Also, check to make sure we are on the psycopg2
account before attempting to rsync the newly created file over.

If on a different account than psycopg2, REMOTE_KEY will not be defined
because the decryption of the key will fail.

dvarrazzo commented:
Hi, this seems perfect: [build worked](https://ci.appveyor.com/project/psycopg/psycopg2-wheels/build/2.x.35) and [libpq versions are right](https://ci.appveyor.com/project/psycopg/psycopg2-wheels/build/2.x.35/job/cwnc6rnf9q2fdshf#L3142). Can we release in this form? Thank you very much!

jerickso committed:
Build pg_config from source and copy include files

Build pg_config from posgtres.
Use Postgres's install include function to copy over install files used
by psycopg.
Force rebuild of AppVeyor cache

jerickso commented:
Hmm, I think my git mucking forced this pull request to pulled in additional changes after I made the request first.  These patches fix the Py3.3/3.4 build problems, as well as fixes the libpq 'version' discrepancy.  They are ready to be merged.

This set of patches does three things:
'fixes' rc.exe so it doesn't fault on running with the x64 version on Py3.3 and 3.4.  The fix is a hack as it copies over the 64bit version (and it's dll) to where the 32bit version is.  I don't like this fix, but I was unable to find where/why it was choosing the 32bit version.  A simple path modification did not fix it.

Added a check on on_success to only upload if being called from the psycopg account (probably only useful for my debugging purposes).

Build and use pg_config based upon the source code used to build libpq.  I tried to follow the standard building procedures of postgres as best I could so integration with future version of postgres would be seemless, but I still had to hack it up some files, so it may break with future versions of postgres.


-jason

dvarrazzo commented:
Thank you very much. Now all the platforms supported build ok with PG10. Will make a new release soon.
